src/SFML/Window/Unix/Utils.hpp

```diff
@@ -0,0 +1,57 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#pragma once
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <X11/Xlib.h>
+
+#include <memory>
+#include <type_traits>
+
+
+namespace sf::priv
+{
+////////////////////////////////////////////////////////////
+/// \brief Class template for freeing X11 pointers
+///
+////////////////////////////////////////////////////////////
+template <class T>
```

Comment:
Note that `XFree` takes a `void*` so a template deleter is not required.

However, maybe we can make use of template specialization or function overloading to support types that are not freed with `XFree` such as `XImage` that is freed with `XDestroyImage`?

Reply:
Oh yeah we can overload that on different X pointer types. That's a great point. Let me play around with that and see what I can do.

Reply:
That wasn't too bad. I specialized the deleter for `XImage`.